### DISCUSSION TITLE
Play nice with others

### DISCUSSION THREAD

**Discussion by MalikRumi** [microsoft/autogen#880] [Q&A]
In light of the changes for v0.2, I want to know if the future includes vendor lock in or a tying arrangement where we will be forced to use an exclusively Microsoft toolset? I hope not. That would be a giant leap backwards, and would stunt the growth of potentially the entire industry.

**Comment by rickyloynd-microsoft** [comment]
Hope not! What aspect of the changes for v0.2 raise a concern for you?

**Reply by MalikRumi** [reply]
What caught my eye was the reliance on either Microsoft (Azure) or OpenAI (again, Microsoft) keys. I have been looking at a lot of new apps and businesses offering AI services of one kind or another, and I have been surprised and disappointed that so many of them are just wrappers around OpenAI models. With new models, both open and closed source, coming out by the hour it seems, I expected to see more entrepreneurs embracing open source models, if for no other reason than cost savings, but that's not what my admittedly research showed me.

So in that context, I am hoping that in the near future Microsoft will offer tools that give the individual developer the freedom to mix and match apps, hosts, and integrations in any way he or she sees fit in order to bring their vision to life.

That will preserve the creativity that stimulates the economy for everyone. Perhaps my fears are not warranted. I would certainly hope so. But those are my concerns, and what led me to post the question. Thank you for taking the time to read and respond. Maybe you can present it to Satya over lunch in the cafeteria tomorrow. 😉

**Reply by AIwasabi** [reply]
I concur with the above. While OpenAI made changes to their API, if one objective is to make autogen LLM agnostic, it would have been helpful to keep some backward compatibility maybe through shims to avoid breaking the implementations people made based on versions of autogen prior to 0.2.

**Reply by kfsone** [reply]
There's a rapidly growing flotilla of LLM apis and I don't think autogen benefits from investing in trying to cope with them all. It would be better for Autogen to support one or two proxies like LiteLLM (for localllms) and OpenRouter (for 3rd party hosted llms like Bard, etc).

**Reply by AIwasabi** [reply]
This would indeed be ideal should we find stable and widely used open source APIs. LiteLLM had more breaking changes than autogen for their v1 when OpenAI updates their API. Openrouter seems amazing! Thanks for the heads up😃
Have you used Openrouter in a project?

**Reply by kfsone** [reply]
I've not yet found an excuse to use OpenRouter, I've been focused on wholly local projects to try and make it possible to "talk" to a 20 year old codebase with hundreds of sub-projects and libraries and different layers of specialization required (build system files like cmake, dependencies, python tooling, CI / build files, DSLs, etc), and trying to figure out how to extinguish some of the superstition emerging around LLMs c/o grifters wanting the easy money to be had.
